### ISSUE TITLE
(false positive) doc_markdown should disregard escaped underscores

### ISSUE THREAD

**Issue by lorepozo** [rust-lang/rust-clippy#2581]
If a docstring has an escaped underscore (`\_`), then clippy should treat that escaped underscore as any other (insignificant) character.

```rust
// lib.rs
//! Suppose a web LaTeX viewer is loaded.
//! (Note that backslashes are escaped for LaTeX.)
//!
//! \\[
//!     \\prod\_{x\\in X} p\_x
//! \\]
```

results in

```
warning: you should put `prod_{x\in` between ticks in the documentation
 --> src/lib.rs:5:10
  |
5 | //!     \\prod\_{x\\in X} p\_x
  |          ^^^^^^^^^^
  |
  = note: #[warn(doc_markdown)] on by default
  = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.190/index.html#doc_markdown

warning: you should put `p_x` between ticks in the documentation
 --> src/lib.rs:5:24
  |
5 | //!     \\prod\_{x\\in X} p\_x
  |                        ^^^
  |
  = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.190/index.html#doc_markdown
```

**Comment by lucasem** [comment]
This was on `clippy-0.0.190`.

**Comment by phansch** [comment]
It looks like this is caused by us using `pulldown_cmark` which is turning

```
\\prod\_{x\\in X} p\_x
```

into

```
\prod_{x\in X} p_x
```

The lint then splits this up into the words `\prod_{x\in X}` and `p_x` which are both causing the lint to trigger.
Due to the backslashes being trimmed by pulldown_cmark, there's nothing we can do about that right now, unfortunately.

**Comment by hellow554** [comment]
This is still an issue today. Any updates on this (after 3 years? :) )
